storage/registers.go

```diff
@@ -0,0 +1,39 @@
+package storage
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RegisterIndex defines methods for the register index.
+type RegisterIndex interface {
+	RegisterIndexReader
+	RegisterIndexWriter
+}
+
+// RegisterIndexReader defines read-only operations on the register index.
+type RegisterIndexReader interface {
+	// Get register by the register ID at a given block height.
+	//
+	// If the register at the given height was not indexed, returns the highest
+	// height the register was indexed at.
+	// An error is returned if the register was not indexed at all or if the height is out of bounds.
+	// Expected errors:
+	// - storage.ErrHeightNotIndexed if the register was not found in the db or is out of bounds.
+	Get(ID flow.RegisterID, height uint64) (flow.RegisterValue, error)
+	// LatestHeight returns the latest indexed height.
+	LatestHeight() (uint64, error)
+	// FirstHeight at which we started to index. Returns the first indexed height found in the store.
+	FirstHeight() (uint64, error)
+}
+
+// RegisterIndexWriter defines write-only operations on the register index.
+type RegisterIndexWriter interface {
+	// Store batch of register entries at the provided block height.
+	//
+	// The provided height must either be one higher than the current height or the same to ensure idempotency,
```

Comment:
Thinking about this again. 

I think we can remove this RegisterIndexWriter interface, and move the `Store` method to `RegisterIndex` interface, because the Writer depends on the Reader method to read the current height. I can't think of a user case where the writer doesn't not depend on the reader.

Reply:
got it